packages/cmr-client/CMR.js

```diff
@@ -0,0 +1,284 @@
+'use strict';
+
+const got = require('got');
+const get = require('lodash.get');
+const publicIp = require('public-ip');
+const Logger = require('@cumulus/logger');
+
+const searchConcept = require('./searchConcept');
+const ingestConcept = require('./ingestConcept');
+const deleteConcept = require('./deleteConcept');
+const getUrl = require('./getUrl');
+const ValidationError = require('./ValidationError');
+
+const log = new Logger({ sender: 'cmr-client' });
+
+const logDetails = {
+  file: 'cmr-client/CMR.js'
+};
+
+/**
+ * Find the UMM version as a decimal string.
+ * If a version cannot be found on the input object
+ * version 1.4 is assumed and returned.
+ *
+ * @param {Object} umm - UMM metadata object
+ * @returns {string} UMM version for the given object
+ *
+ * @private
+ */
+function ummVersion(umm) {
+  return get(umm, 'MetadataSpecification.Version', '1.4');
+}
+
+/**
+ * Posts a given xml string to the validate endpoint of CMR
+ * and promises true of valid.
+ *
+ * @param {string} ummMetadata - the UMM object
+ * @param {string} identifier - the document identifier
+ * @param {string} provider - the CMR provider
+ * @returns {Promise<boolean>} returns true if the document is valid
+ *
+ * @private
+ */
+async function validateUMMG(ummMetadata, identifier, provider) {
+  const version = ummVersion(ummMetadata);
+  let result;
+
+  try {
+    result = await got.post(`${getUrl('validate', provider)}granule/${identifier}`, {
+      json: true,
+      body: ummMetadata,
+      headers: {
+        Accept: 'application/json',
+        'Content-type': `application/vnd.nasa.cmr.umm+json;version=${version}`
+      }
+    });
+
+    if (result.statusCode === 200) {
+      return true;
+    }
+  }
+  catch (e) {
+    result = e.response;
+  }
+
+  throw new ValidationError(
+    `Validation was not successful. UMM metadata Object: ${JSON.stringify(ummMetadata)}`
+  );
+}
+
+/**
+ * Returns a valid a CMR token
+ *
+ * @param {string} cmrProvider - the CMR provider id
+ * @param {string} clientId - the CMR clientId
+ * @param {string} username - CMR username
+ * @param {string} password - CMR password
+ * @returns {Promise.<string>} the token
```

Comment:
Looking at the [JSDoc documentation](http://usejsdoc.org/tags-type.html), it looks like they use `Array.<string>` for specifying the type stored in an Array.  It would make sense to use the same syntax for Promises.  `Promise.<string>`

After further research, it looks like usejsdoc.org also uses [Promise\<string\>](http://usejsdoc.org/tags-async.html) so ...